admin/admin.go

```diff
@@ -145,6 +160,32 @@ func (m *RPCMethods) GetL1Addresses(args *uint64) *map[string]string {
 		"ProxyAdmin":                        chain.L2Config.L1Addresses.ProxyAdmin.String(),
 		"SuperchainConfig":                  chain.L2Config.L1Addresses.SuperchainConfig.String(),
 	}
-	m.Log.Debug("admin_getL2Addresses")
+	m.log.Debug("admin_getL2Addresses")
 	return &reply
 }
+
+func (m *RPCMethods) GetL2ToL2MessageByMsgHash(args *common.Hash) *interop.L2ToL2Message {
+
+	if args == nil {
+		m.log.Error("valid msg hash not provided", "args", *args)
+		return nil
+	}
+
+	if m.networkConfig.InteropEnabled && m.l2ToL2MsgIndexer != nil {
+
+		storeEntry, err := m.l2ToL2MsgIndexer.Get(*args)
```

Comment:
also need to handle case if the storeEntry just doesn't exist - should end up returning nil